### PR TITLE
Partially fix columns and viz settings mismatch

### PR DESCRIPTION
**Pull request by uladzimirdev**
attempted to fix 
https://github.com/metabase/metabase/issues/42049, but it didn't work

needs to wait for more general solution which will unify keys between viz settings and cols

### Description

PR in current state ignores base-type from field ref to exclude BE misalignment

### How to verify

- all tests should pass

### Checklist

- [x] added a repro for the linked issue

### CODE REVIEW

File: e2e/test/scenarios/question/settings.cy.spec.js [hunk 1]
```diff
@@ -137,8 +137,21 @@ describe("scenarios > question > settings", () => {
                 "source-table": PRODUCTS_ID,
                 condition: [
                   "=",
-                  ["field-id", ORDERS.PRODUCT_ID],
-                  ["joined-field", "Products", ["field-id", PRODUCTS.ID]],
+                  [
```
Comment (uladzimirdev) [inline e2e/test/scenarios/question/settings.cy.spec.js:140]: current format

**Review by ranquild — APPROVED** [review]
@uladzimirdev could you create an e2e test that verifies that `ignoreBaseType` is present? maybe similar to the referenced issue

**Comment by uladzimirdev** [comment]
@ranquild I can't think of any test, do you have an idea? 
I realized that previously (without ignoreBaseType) it worked in the same way as we call `syncTableColumnSettings`twice` and first time columns that are not fined in the settings are mapped with that `addedColumns` 

```
 "table.columns": [...existingColumnSettings, ...addedColumnSettings],
```

and later on when the second rendering is happening, all the mappings are correct and contain `base-type`, so it's "fixed"

**Comment by uladzimirdev** [comment]
@ranquild added unit tests

**Review by ranquild — APPROVED** [review]

**Pull request merged**